src/batch.js

```diff
@@ -83,30 +83,57 @@ function handleBatch(router, req) {
     req.config.publicServerURL
   );
 
-  const promises = req.body.requests.map(restRequest => {
-    const routablePath = makeRoutablePath(restRequest.path);
-    // Construct a request that we can send to a handler
-    const request = {
-      body: restRequest.body,
-      config: req.config,
-      auth: req.auth,
-      info: req.info,
-    };
+  let initialPromise = Promise.resolve();
+  if (req.body.transaction === true) {
+    initialPromise = req.config.database.createTransactionalSession();
```

Comment:
Good job getting this one in. I can't wait to use it.

Can you add this to direct access?

https://github.com/parse-community/parse-server/blob/master/src/ParseServerRESTController.js

Reply:
done